src/service/artifacts/upload-service.ts

```diff
@@ -142,23 +141,23 @@ export class UploadService {
     }
   }
 
-  async upload(archiveArtifacts: ArchiveArtifacts): Promise<artifact.UploadResponse> {
+  async upload(archiveArtifacts: ArchiveArtifacts, projectName: string): Promise<artifact.UploadResponse> {
     // remove the filter once build-chain-config reader is refactored
     const searchPaths = archiveArtifacts.paths.filter(pathItem => pathItem.path).reduce((prev: string, curr) => prev.concat(curr.path!, "\n"), "");
 
     const { filesToUpload, rootDirectory } = await this.findFilesToUpload(searchPaths);
     if (filesToUpload.length === 0) {
       this.noFileFound(archiveArtifacts, searchPaths);
       return {
-        artifactName: archiveArtifacts.name,
+        artifactName: archiveArtifacts.name ?? projectName,
```

Comment:
what if you provide 
```
const artifactName = archiveArtifacts.name ?? projectName
```
and
```suggestion
        artifactName,
```

this way you modify L160 as well, wdyt?